public/locations-list-map.js

```diff
@@ -768,23 +770,32 @@ function centerMapToMarkersNearUser() {
 function fitMapToMarkersNearBounds(bounds) {
   // get center of bounding box and use it to sort markers by distance
   let center = bounds.getCenter();
-  const markersByDistance = getMarkersByDistanceFrom(center.lat(), center.lng());
-  
+  const markersByDistance = getMarkersByDistanceFrom(center.lat(), center.lng(), 3);
+
+  let hasMarker = false;
+
   // extend bounds to fit closest three markers
   [0,1,2].forEach((i) => {
     const marker = markersByDistance[i];
     if (marker) {
+      hasMarker = true;
       bounds.extend(marker.position);
     }
   });
 
-  map.fitBounds(bounds);
+  if (hasMarker) {
```

Comment:
I wouldn't do this. I'd check bounds.getNorthEast and bounds.getSouthWest and see if they're different. If so, call fitBounds. If not, call setCenter.